### PR TITLE
HG-653 escape query params before inserting into document

### PR DESCRIPTION
**Pull request by kenkouot**
@Grunny

### CODE REVIEW

**Comment by rogatty** [comment]
Wouldn't a whitelist be a better solution? For sure it would be safer.

**Comment by kenkouot** [comment]
It would be a better solution and is handled natively in Hapi's validation if we went that path, but @hobbs seems to think that the variability of our query parameters (some have random ids appended to the keys) makes that a non-starter/

**Comment by hakubo** [comment]
Right we also have to think if we still need that part of code that reflects all queryparams.

It's LGTM for the fix, but I would think if we could remove that part at all.

**Comment by rogatty** [comment]
Currently we use `M.prop('queryParams')` in only two places which are:
https://github.com/Wikia/mercury/blob/dev/front/scripts/mercury/utils/track.ts#L116 and https://github.com/Wikia/mercury/blob/dev/front/scripts/mercury/utils/track.ts#L156 both looking like this:
```
if (M.prop('queryParams.noexternals')) {
	return;
}
```

Maybe this should be solved in another way?

**Comment by rogatty** [comment]
LGTM for a quick P2 fix.

File: server/views/_helpers/escapeQueryParams.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	var keys = Object.keys(query),
+		obj = {};
+
+	keys.forEach((key) => {
+		obj[Hoek.escapeHtml(key)] = Hoek.escapeHtml(query[key].toString());
+	});
+
+	return obj;
+}
+export = decodeQueryParams;
```
Comment (rogatty) [inline server/views/_helpers/escapeQueryParams.ts]: File name is `escapeQueryParams.ts`. Should it be consistent?

**Comment by Grunny** [comment]
LGTM as a quick fix. I'm not familiar enough with Mercury yet, but would it be possible to never store query params in a script tag within the document anyway as it is user controlled data? Can they just be retrieved either on demand or retrieved at the start client-side?

**Comment by kenkouot** [comment]
I created HG-655 as a followup to this

**Comment by kenkouot** [comment]
Gonna merge this as a quick fix once tests pass, the root cause should be addressed in our next "sprint"

**Pull request merged**